src/ai/backend/manager/api/gql/background_task.py

```diff
@@ -0,0 +1,226 @@
+from __future__ import annotations
+
+import logging
+import uuid
+from datetime import datetime
+from typing import AsyncGenerator, Optional
+
+import strawberry
+from strawberry import ID
+from strawberry.relay import Node, NodeID
+from strawberry.types import Info
+
+from ai.backend.common.events.hub.propagators.cache import WithCachePropagator
+from ai.backend.common.events.types import EventCacheDomain, EventDomain
+from ai.backend.logging import BraceStyleAdapter
+
+from .types import StrawberryGQLContext
+
+log = BraceStyleAdapter(logging.getLogger(__spec__.name))
+
+
+@strawberry.type
+class BackgroundTaskEvent(Node):
+    """Background task event information"""
+
+    id: NodeID[str]
+    task_id: ID
+    event_name: str
+    data: strawberry.scalars.JSON
+    timestamp: datetime
+    retry_count: Optional[int] = None
+
+
+@strawberry.type
+class BackgroundTaskProgress(Node):
+    """Background task progress information"""
+
+    id: NodeID[str]
+    task_id: ID
+    current_progress: Optional[float] = None
+    total_progress: Optional[float] = None
+    message: Optional[str] = None
```

Comment:
I think it would be enough to just send the information for `pending_count`, `success_count`, `failure_count`, and `total_count`.